File: src/main/java/com/dawdawich/utils/TelegramAd.java
package com.dawdawich.utils;

import org.telegram.telegrambots.api.methods.PartialBotApiMethod;
import org.telegram.telegrambots.api.methods.send.SendDocument;
import org.telegram.telegrambots.api.methods.send.SendMessage;
import org.telegram.telegrambots.api.methods.send.SendPhoto;
import org.telegram.telegrambots.api.objects.Document;
import org.telegram.telegrambots.api.objects.Message;
import org.telegram.telegrambots.api.objects.PhotoSize;
import org.telegram.telegrambots.api.objects.replykeyboard.InlineKeyboardMarkup;
import org.telegram.telegrambots.api.objects.replykeyboard.buttons.InlineKeyboardButton;

import java.util.ArrayList;
import java.util.List;

public class TelegramAd {

    List<InlineKeyboardButton> keyboardButtons = new ArrayList<>();
    private String description;
    private int timeZone;
    private Integer hour;
    private Integer minute;
    private int intervalTop;
    private int interval;
    private PhotoSize photo;
    private Document document;
    private long chatId;
    private int adId;
    private boolean isActive = false;
    private PartialBotApiMethod<Message> newAd;

    public void setDescription(String description) {
        this.description = description;
    }

    public void setTimeZone(int timeZone) {
        this.timeZone = timeZone;
    }

    public void setHour(Integer hour) {
        this.hour = hour;
    }

    public void setMinute(Integer minute) {
        this.minute = minute;
    }

    public void setIntervalTop(int intervalTop) {
        this.intervalTop = intervalTop;
    }

    public void setInterval(int interval) {
        this.interval = interval;
    }

    public void setPhoto(PhotoSize photo) {
        this.photo = photo;
    }

    public void setDocument(Document document) {
        this.document = document;
    }

    public List<InlineKeyboardButton> getKeyboardButtons() {
        return keyboardButtons;
    }

    public void setChatId(long chatId) {
        this.chatId = chatId;
    }

    public int getIntervalTop() {
        return intervalTop;
    }

    public int getInterval() {
        return interval;
    }

    public int getAdId() {
        return adId;
    }

    public void setAdId(int adId) {
        this.adId = adId;
    }

    public boolean isActive() {
        return isActive;
    }

    public void setActive(boolean active) {
        isActive = active;
    }

    public int getTimeZone() {
        return timeZone;
    }

    public Integer getHour() {
        return hour;
    }

    public Integer getMinute() {
        return minute;
    }

    public PartialBotApiMethod<Message> getNewAd() {
        return newAd;
    }

    public void earmarkedAd() {
        InlineKeyboardMarkup markup = new InlineKeyboardMarkup();
        for (InlineKeyboardButton button : keyboardButtons) {
            markup.getKeyboard().add(new ArrayList<InlineKeyboardButton>() {{
                add(button);
            }});
        }
        if (photo != null) {
            newAd = new SendPhoto();
            ((SendPhoto) newAd).setChatId(chatId);
            ((SendPhoto) newAd).setPhoto(photo.getFileId());
            ((SendPhoto) newAd).setCaption(description);
            ((SendPhoto) newAd).setReplyMarkup(markup);
        } else if (document != null) {
            newAd = new SendDocument();
            ((SendDocument) newAd).setChatId(chatId);
            ((SendDocument) newAd).setDocument(document.getFileId());
            ((SendDocument) newAd).setCaption(description);
            ((SendDocument) newAd).setReplyMarkup(markup);
        } else {
            newAd = new SendMessage();
            ((SendMessage) newAd).setChatId(chatId);
            ((SendMessage) newAd).setText(description);
            ((SendMessage) newAd).setReplyMarkup(markup);
        }
        EarmarkedPost.setAd(this);
    }

}

File: src/main/java/com/dawdawich/utils/TelegramAd.kt
package com.dawdawich.utils

import org.telegram.telegrambots.api.methods.PartialBotApiMethod
import org.telegram.telegrambots.api.methods.send.SendDocument
import org.telegram.telegrambots.api.methods.send.SendMessage
import org.telegram.telegrambots.api.methods.send.SendPhoto
import org.telegram.telegrambots.api.objects.Document
import org.telegram.telegrambots.api.objects.Message
import org.telegram.telegrambots.api.objects.PhotoSize
import org.telegram.telegrambots.api.objects.replykeyboard.InlineKeyboardMarkup
import org.telegram.telegrambots.api.objects.replykeyboard.buttons.InlineKeyboardButton

import java.util.ArrayList

class TelegramAd {

    var keyboardButtons: List<InlineKeyboardButton> = ArrayList()
        internal set
    private var description: String? = null
    var timeZone: Int = 0
    var hour: Int? = null
    var minute: Int? = null
    var intervalTop: Int = 0
    var interval: Int = 0
    private var photo: PhotoSize? = null
    private var document: Document? = null
    private var chatId: Long = 0
    var adId: Int = 0
    var isActive = false
    var newAd: PartialBotApiMethod<Message>? = null
        private set

    fun setDescription(description: String) {
        this.description = description
    }

    fun setPhoto(photo: PhotoSize) {
        this.photo = photo
    }

    fun setDocument(document: Document) {
        this.document = document
    }

    fun setChatId(chatId: Long) {
        this.chatId = chatId
    }

    fun earmarkedAd(earmarkedPost: EarmarkedPost) {
        val markup = InlineKeyboardMarkup()
        for (button in keyboardButtons) {
            markup.keyboard.add(object : ArrayList<InlineKeyboardButton>() {
                init {
                    add(button)
                }
            })
        }
        when {
            photo != null -> {
                newAd = SendPhoto()
                (newAd as SendPhoto).setChatId(chatId)
                (newAd as SendPhoto).photo = photo!!.fileId
                (newAd as SendPhoto).caption = description
                (newAd as SendPhoto).replyMarkup = markup
            }
            document != null -> {
                newAd = SendDocument()
                (newAd as SendDocument).setChatId(chatId)
                (newAd as SendDocument).document = document!!.fileId
                (newAd as SendDocument).caption = description
                (newAd as SendDocument).replyMarkup = markup
            }
            else -> {
                newAd = SendMessage()
                (newAd as SendMessage).setChatId(chatId)
                (newAd as SendMessage).text = description
                (newAd as SendMessage).replyMarkup = markup
            }
        }
        earmarkedPost.setAd(this)
    }

}

